projects/packages/sync/src/modules/class-module.php

```diff
@@ -338,23 +360,31 @@ public function send_full_sync_actions( $config, $status, $send_until ) {
 		$limits = Settings::get_setting( 'full_sync_limits' )[ $this->name() ];
 
 		$chunks_sent = 0;
-		// phpcs:ignore Generic.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition
-		while ( $objects = $this->get_next_chunk( $config, $status, $limits['chunk_size'] ) ) {
-			if ( $chunks_sent++ === $limits['max_chunks'] || microtime( true ) >= $send_until ) {
+
+		$last_item = $this->get_last_item( $config );
+
+		while ( $chunks_sent < $limits['max_chunks'] && microtime( true ) < $send_until ) {
+			$objects = $this->get_next_chunk( $config, $status, $limits['chunk_size'] );
+
+			if ( $wpdb->last_error ) {
 				return $status;
```

Comment:
```suggestion
                                 $status['error'] = true;
				return $status;
```
Shouldn't we also update the `$status` here?